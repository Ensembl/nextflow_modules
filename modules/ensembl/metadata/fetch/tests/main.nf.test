// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// nf-core modules test metadata/fetch
nextflow_process {

    name "Test Process METADATA_FETCH"
    script "../main.nf"
    process "METADATA_FETCH"

    tag "metadata"
    tag "metadata/fetch"

    test("Metadata fetch test [real test]") {
        
        when {
            process {
                """
                input[0] = [ 
                    [accession: 'GCA_000001215.4'], 
                    file("${projectDir}/tests/modules/ensembl/metadata/fetch/meta-query.json", checkIfExists: true),
                    file("${projectDir}/tests/modules/ensembl/metadata/fetch/ncbi_genome-summary.json", checkIfExists: true)
                    ]
                """
            }
        }

            then {
                assert snapshot(
                    process.out.meta_json.collect {
                        it.collect { it instanceof String && file(it).exists() ? file(it).name : it }
                        }
                    ).match("GCA_000001215.4-genome-meta.json")

            assertAll(
                    { assert process.success },
                    { assert snapshot(process.out).match() },
                    { assert process.out.meta_json.size() == 1 },
                    { assert process.out.versions != null }
            )
        }
    }

    test("Metadata fetch test [stub]") {
        
        when {
            options "-stub"
            process {
                """
                input[0] = [ 
                    [accession: 'GCA_000111222.3'], 
                    file("query.json"),
                    file("ncbi-genome.json")
                    ]
                """
                    // file("${projectDir}/tests/modules/ensembl/metadata/fetch/meta-query.json", checkIfExists: true),
                    // file("${projectDir}/tests/modules/ensembl/metadata/fetch/ncbi_genome-summary.json", checkIfExists: true)
            }
        }

            then {
                assert snapshot(
                    process.out.meta_json.collect {
                        it.collect { it instanceof String && file(it).exists() ? file(it).name : it }
                        }
                    ).match("GCA_000111222.3-genome-meta.json")

            assertAll(
                    { assert process.success },
                    { assert snapshot(process.out).match() },
                    { assert process.out.meta_json.size() == 1 },
                    { assert process.out.versions != null }
            )
        }
    }
}
