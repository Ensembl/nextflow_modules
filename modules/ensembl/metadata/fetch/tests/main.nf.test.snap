{
    "GCA_000111222.3-genome-meta.json": {
        "content": [
            [
                [
                    {
                        "accession": "GCA_000111222.3"
                    },
                    "GCA_000111222.3-genome-meta.json"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-24T15:31:12.678842429"
    },
    "Metadata fetch test [real test]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "accession": "GCA_000001215.4"
                        },
                        "GCA_000001215.4-genome-meta.json:md5,2ac66ff0b5c14c2b910122f485f7ca05"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a8f1ba0ddf1d46514d70db64d61f470b"
                ],
                "meta_json": [
                    [
                        {
                            "accession": "GCA_000001215.4"
                        },
                        "GCA_000001215.4-genome-meta.json:md5,2ac66ff0b5c14c2b910122f485f7ca05"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a8f1ba0ddf1d46514d70db64d61f470b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-24T15:26:56.308213972"
    },
    "GCA_000001215.4-genome-meta.json": {
        "content": [
            [
                [
                    {
                        "accession": "GCA_000001215.4"
                    },
                    "GCA_000001215.4-genome-meta.json"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-24T15:26:56.226059575"
    },
    "Metadata fetch test [stub]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "accession": "GCA_000111222.3"
                        },
                        "GCA_000111222.3-genome-meta.json:md5,35290e51b83617b0deb85f4da6fd0b51"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a8f1ba0ddf1d46514d70db64d61f470b"
                ],
                "meta_json": [
                    [
                        {
                            "accession": "GCA_000111222.3"
                        },
                        "GCA_000111222.3-genome-meta.json:md5,35290e51b83617b0deb85f4da6fd0b51"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a8f1ba0ddf1d46514d70db64d61f470b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-24T15:31:12.753622423"
    }
}